source/common/common/logger.h

```diff
@@ -123,48 +195,16 @@ template <Id id> class Loggable {
 
 } // Logger
 
-#ifdef NDEBUG
-#define log_trace(...)
-#define log_debug(...)
-#else
-#define log_trace(...) log().trace(__VA_ARGS__)
-#define log_debug(...) log().debug(__VA_ARGS__)
-#endif
-
-/**
- * Convenience macros for logging with connection ID.
- */
-#define conn_log(LOG, LEVEL, FORMAT, CONNECTION, ...)                                              \
-  LOG.LEVEL("[C{}] " FORMAT, (CONNECTION).id(), ##__VA_ARGS__)
-
-#ifdef NDEBUG
-#define conn_log_trace(...)
-#define conn_log_debug(...)
-#else
-#define conn_log_trace(FORMAT, CONNECTION, ...)                                                    \
-  conn_log(log(), trace, FORMAT, CONNECTION, ##__VA_ARGS__)
-#define conn_log_debug(FORMAT, CONNECTION, ...)                                                    \
-  conn_log(log(), debug, FORMAT, CONNECTION, ##__VA_ARGS__)
-#endif
-
-#define conn_log_info(FORMAT, CONNECTION, ...)                                                     \
-  conn_log(log(), info, FORMAT, CONNECTION, ##__VA_ARGS__)
-
-/**
- * Convenience macros for logging with a stream ID and a connection ID.
- */
-#define stream_log(LOG, LEVEL, FORMAT, STREAM, ...)                                                \
-  LOG.LEVEL("[C{}][S{}] " FORMAT, (STREAM).connectionId(), (STREAM).streamId(), ##__VA_ARGS__)
-
-#ifdef NDEBUG
-#define stream_log_trace(...)
-#define stream_log_debug(...)
-#else
-#define stream_log_trace(FORMAT, STREAM, ...)                                                      \
-  stream_log(log(), trace, FORMAT, STREAM, ##__VA_ARGS__)
-#define stream_log_debug(FORMAT, STREAM, ...)                                                      \
-  stream_log(log(), debug, FORMAT, STREAM, ##__VA_ARGS__)
-#endif
-
-#define stream_log_info(FORMAT, STREAM, ...) stream_log(log(), info, FORMAT, STREAM, ##__VA_ARGS__)
+// Function for formatting the connection log
+template <typename T, typename... Args>
+std::string format_connection_log(const std::string& format, T& connection, Args&&... args) {
```

Comment:
I'm not sure if these should be macros or regular functions.  A macro might make it a bit faster by concatenating the two string literals at compile time rather than forcing the creation of two string objects, but that might not be a concern.